core/vm/runtime/runtime.hpp

```diff
@@ -141,6 +142,59 @@ namespace fc::vm::runtime {
     /// Update actor state CID
     virtual outcome::result<void> commit(const ActorSubstateCID &new_state) = 0;
 
+    /// Resolve address to id-address
+    virtual outcome::result<Address> resolveAddress(const Address &address) = 0;
+
+    /// Verify PoSt
+    virtual outcome::result<bool> verifyPoSt(uint64_t sector_size,
+                                             const PoStVerifyInfo &info) = 0;
+
+    /// Verify seal
+    virtual outcome::result<bool> verifySeal(uint64_t sector_size,
+                                             const SealVerifyInfo &info) = 0;
+
+    /// Send funds
+    inline auto sendFunds(const Address &to, BigInt value) {
+      // kSendMethodNumber circular dependency
+      return send(to, {0}, {}, value);
+    }
+
+    /// Send with typed result R
+    template <typename R>
+    outcome::result<R> sendR(Address to_address,
+                             MethodNumber method_number,
+                             const MethodParams &params,
+                             BigInt value) {
+      OUTCOME_TRY(result, send(to_address, method_number, params, value));
+      return codec::cbor::decode<R>(result.return_value);
+    }
+
+    /// Send with typed params P and result R
+    template <typename R, typename P>
+    outcome::result<R> sendPR(Address to_address,
+                              MethodNumber method_number,
+                              const P &params,
+                              BigInt value) {
+      OUTCOME_TRY(params2, codec::cbor::encode(params));
+      return sendR<R>(to_address, method_number, MethodParams{params2}, value);
+    }
+
+    /// Send with typed params P
+    template <typename P>
+    outcome::result<InvocationOutput> sendP(Address to_address,
+                                            MethodNumber method_number,
+                                            const P &params,
+                                            BigInt value) {
+      OUTCOME_TRY(params2, codec::cbor::encode(params));
+      return send(to_address, method_number, MethodParams{params2}, value);
+    }
+
+    /// Get decoded current actor state
+    template <typename T>
+    outcome::result<T> getCurrentActorStateCbor() {
```

Comment:
```suggestion
    outcome::result<T> getCurrentActorState() {
```

Reply:
Still need way to get cid, when type is unknown